### PR TITLE
replaced cssmin with PyScss for css compression.

### PR DESCRIPTION
**Pull request by loechel**
Replaced cssmin with PyScss to ensure Python 3 compatibility and maintainability. 
Removed dependency to cssmin, so could break dependency for third party addons that depend on it.
Introduced PyScss as a drop in replacement that could also do more things.
Discussion on that at https://github.com/plone/Products.CMFPlone/issues/1800
Pull request requested by @gforcada

### CODE REVIEW

**Comment by loechel** [comment]
@gforcada did you see why the Jenkins build failed? there is not enough scroll back, but it seems like the PyScss module is not installed during buildout.

**Comment by gforcada** [comment]
@loechel I hoped you had an idea already, I just saw the error but did not dig further.

As for the full log, if you look closely, at the very top of http://jenkins.plone.org/job/pull-request-5.1/299/console there is a link to http://jenkins.plone.org/job/pull-request-5.1/299/consoleFull

**Comment by loechel** [comment]
Might it be that the pull request did not see the branch at buildout time? so that we need something like a plip.cfg in buildout.coredev to run the buildout against?

**Comment by loechel** [comment]
@gforcada base problem fixed, but there seems to be more errors.

I don't see why the Pull request also tries to load tests for pyScss.
Also there is a UnicodeDecodeError for an mdash which I don't see.

**Comment by gforcada** [comment]
Couldn't be that this dash is coming from the CSS and pyscss needs to have the data encoded?

**Comment by loechel** [comment]
ok @gforcada I have found the error and fixed it. It was actually the other way around pyscss accepts unicode, but also returns unicode and the cooked_css only accepts ascii strings

tests for css minification run locally. 
But test suite is failing again on Products.CMFPlone.tests.testMigrationTool.TestMigrationTool and Products.CMFPlone.tests.test_factory.TestFactoryPloneSite. Should we ignore that?

**Comment by loechel** [comment]
@gforcada anything else on that pull request? Did you need something more to merge?

File: Products/CMFPlone/resources/browser/cook.py [hunk 1]
```diff
@@ -90,28 +92,30 @@ def cookWhenChangingSettings(context, bundle=None):
                 css_url = siteUrl + '/' + css_resource
                 response = subrequest(css_url)
                 if response.status == 200:
-                    logger.info('Cooking css {0}'.format(css_resource))
+                    logger.info('Cooking css %s', css_resource)
                     css = response.getBody()
-                    cooked_css += '\n/* Resource: {0} */\n{1}\n'.format(
+                    if css_resource[-8:] != '.min.css':
+                        if sys.version_info == (2):
```
Comment (gforcada) [inline Products/CMFPlone/resources/browser/cook.py]: This conditional returns ``False`` on my python 2 prompt, but the comment on the line below implies that it should be true... maybe you wanted to write ``sys.version_info[0] == 2`` ?

**Comment by loechel** [comment]
@gforcada you are right, with the general translation into unicode that check wasn't even necessary anymore. I did remove it. I hope it is now ready for merge.

**Comment by gforcada** [comment]
*while tests are running*

@loechel would you mind adding the version pins needed on buildout.coredev? I see

```
# Required by:
# pyScss==1.3.5
pathlib = 1.0.1

# Required by:
# Products.CMFPlone==5.1b5.dev0
pyScss = 1.3.5
```

on the jenkins output

**Comment by loechel** [comment]
@gforcada done in https://github.com/plone/buildout.coredev/pull/355

**Pull request merged**